domain_models/models.py

```diff
@@ -211,3 +212,11 @@ def get_data(self):
         return dict((name, field.get_builtin_type(self))
                     for name, field in
                     six.iteritems(self.__class__.__fields__))
+
+    def set_data(self, data):
+        """Set dictionary data to model.
+
+        :param dict data:
+        """
+        for name, value in six.iteritems(data):
+            self.__class__.__fields__[name].set_builtin_type(self, value)
```

Comment:
I have chosen variant with iteration of input data because of it is more useful imo. E.g. some model has three fields and one of those is not required.
```
class Profile(models.DomainModel):
    id = fields.Int()
    name = fields.Str()
    subscriber = fields.Bool(default=True)
``` 
If we pass `{id: 1, name: 'John'}` as argument for `set_data` and iterate this one dict we have successfully filled out model. In other case (with iteration of fields) we have an exception raised or we have to implement some checking whether current field is required or not.

That is why I have chosen iteration of dict. What do you think?